rwengine/src/render/OpenGLRenderer.hpp

```diff
@@ -341,38 +347,34 @@ class OpenGLRenderer : public Renderer {
     // State Cache
     DrawBuffer* currentDbuff = nullptr;
     OpenGLShaderProgram* currentProgram = nullptr;
-    bool blendEnabled = false;
-    size_t blendMode = 0;
+    BlendMode blendMode = BLEND_NONE;
     bool depthWriteEnabled = false;
     GLuint currentUBO = 0;
     GLuint currentUnit = 0;
     std::map<GLuint, GLuint> currentTextures;
 
-    void setBlendMode(size_t mode) {
-        switch (mode) {
-            default:
-            case 0:
-                glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
-                break;
-            case 1:
-                glBlendFunc(GL_ONE, GL_ONE);
-                break;
-        }
-        blendMode = mode;
-    }
-
     // Set state
-    void setBlend(bool enable, size_t mode) {
-        if (enable && !blendEnabled) {
+    void setBlend(BlendMode mode) {
+        if (mode!=BLEND_NONE && blendMode==BLEND_NONE)//To don't call glEnable again when it already enabled
             glEnable(GL_BLEND);
-            setBlendMode(mode);
-            blendEnabled = enable;
-        } else if (enable && blendEnabled && blendMode!=mode) {
-            setBlendMode(mode);
-        } else if(!enable && blendEnabled) {
-            glDisable(GL_BLEND);
-            blendEnabled = enable;
+
+        if (mode!=blendMode) {
+            switch (mode) {
+                default:
+                case BLEND_NONE:
+                    glDisable(GL_BLEND);
+                    break;
+                case BLEND_ALPHA:
+                    glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
+                    break;
+                case BLEND_ADDITIVE:
+                    glBlendFunc(GL_ONE, GL_ONE);
+                    break;
+
```

Comment:
I'd rather see

```c
default:
  assert(false);
  break;
```

We'll have undefined behaviour anyway in that case - so might as well catch it in the debugger